### PR TITLE
Optimize conformer generation

### PR DESCRIPTION
**Pull request by mariusrueve**
I found this [blog post](https://greglandrum.github.io/rdkit-blog/posts/2022-09-29-optimizing-conformer-generation-parameters.html) and, for now, added the "optimized" conformer generation.

Feel free to give more points for optimization if needed.

Lost the overview, but @nikplx are you involved in conformer generation? If so also give feedback

### CODE REVIEW

**Comment by nikplx** [comment]
lol I did actually basically the same thing locally today when trying to fine tune the conformer embedder :D

File: src/lib/coaler/embedder/ConformerEmbedder.cpp [hunk 1]
```diff
@@ -29,14 +30,16 @@ namespace coaler::embedder {
...
+        for (auto const &match: matches) {
             CoreAtomMapping molQueryCoords;
-            for (const auto& [queryId, molId] : match) {
+            for (const auto &[queryId, molId]: match) {
                 molQueryCoords[molId] = m_coords.at(queryId);
             }
 
             // embed molecule conformers
             RDKit::DGeomHelpers::EmbedParameters params;
+            params = RDKit::DGeomHelpers::ETKDGv3;
```
Comment (nikplx) [inline src/lib/coaler/embedder/ConformerEmbedder.cpp:41]: I guess we should use `srETKDGv3` as this appears to be an optimized version

**Review by nikplx — APPROVED** [review]

File: src/lib/coaler/embedder/ConformerEmbedder.cpp (hunk 1 above)
Comment (mariusrueve) [reply src/lib/coaler/embedder/ConformerEmbedder.cpp:41]: `srETKDGv3()` just adds SmallRingTorsion if I understand correctly
```
   # this includes addtional small ring torsion potentials
   params = AllChem.srETKDGv3()

   # this includes additional macrocycle ring torsion potentials and macrocycle-specific handles
   params = AllChem.ETKDGv3()

   # to use the two in conjunction, do:
   params = AllChem.ETKDGv3()
   params.useSmallRingTorsions = True

   # a macrocycle attached to a small ring
   mol = Chem.MolFromSmiles("C(OC(CCCCCCC(OCCSC(CCCCCC1)=O)=O)OCCSC1=O)N1CCOCC1")
   mol = Chem.AddHs(mol)
   AllChem.EmbedMultipleConfs(mol, numConfs = 3 , params = params)
```
From [this](https://github.com/rdkit/rdkit/blob/9c1d1a84f4798b23cba54daf9739e5e6cb2c9b29/Docs/Book/Cookbook.rst#L2427)

Comment (mariusrueve) [reply src/lib/coaler/embedder/ConformerEmbedder.cpp:41]: Resolved in 7fc13ef

**Pull request merged**